# CIS530 Final Project Assignment
**Name:** _[Insert your name here]_

Welcome to your individual assignment based on the *Dig* dataset from the book *Python for MBAs*. You will analyze the Dig case data to answer a unique business question using Python and pandas.

This assignment is worth 25 points and will 'count' for your Quizzes 5-8 in the course.

> 📝 **Instructions:**
>
> - Replace placeholders with your assigned business question.
> - Include comments in your code to explain what each step does.
> - Submit your completed notebook as a `.ipynb` file the Final Project Assignment link in Brightspace

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


## 📌 Your Assigned Question
> Replace the text below with **your specific question** from the assignment document:

**Question**: _[Your unique business question goes here]_  

## 📂 Load and Inspect the Data

In this section, you will load one (or more) of the four key data sets:
- `Restaurants.csv`
- `Items.csv`
- `Simplified orders.csv`
- `Summarized orders.csv`

You can just import the data sets you need for your specific problem. You do not have to import any data that you do not need.

In [ ]:
import pandas as pd
df_orders = pd.read_csv("/content/drive/My Drive/CIS530 Data/Simplified orders.csv")
df_summarized_orders = pd.read_csv("/content/drive/My Drive/CIS530 Data/Summarized orders.csv")
df_restaurants=pd.read_csv("/content/drive/My Drive/CIS530 Data/Restaurants.csv")
df_items=pd.read_csv("/content/drive/My Drive/CIS530 Data/Items.csv")
df_simplified_orders=pd.read_csv("/content/drive/My Drive/CIS530 Data/Simplified orders.csv")


## 📈 Analysis 10 points

This is the core of your work. Use appropriate pandas operations, filters, aggregations,visualizations, and any data frames that need to be merged to answer your assigned question.

***Before you Begin:*** Only use Python or pandas functions from Chapters 5-7 in our text Python for MBAs. If you use any other functions or code that is not specifically from our text, a minimum of 5 points/marks will be deducted for this part of the assignment.

In [ ]:
# your analysis code goes here
#1.Checking the columns
df_orders.columns

# Filter orders where the DRINKS column is >= 1 (at least one drink)
drink_orders = df_orders[df_orders['DRINKS'] >= 1]
#Drink Upsells
drink_upsells = len(drink_orders)
print(drink_upsells)

# Filter orders with at least one cookie
cookie_orders = df_orders[df_orders['COOKIES'] >= 1]

#Cookie Upsells
cookie_upsells = len(cookie_orders)
print(cookie_upsells)

# Group by RESTAURANT_ID and count the number of orders with cookies
drink_orders_per_restaurant = drink_orders.groupby('RESTAURANT_ID').size().sort_values(ascending=False)
print(drink_orders_per_restaurant)


# Group by RESTAURANT_ID and count the number of orders with cookie
cookie_orders_per_restaurant = cookie_orders.groupby('RESTAURANT_ID').size().sort_values(ascending=False)
print(cookie_orders_per_restaurant)



# Calculate the total number of orders
total_orders = len(df_orders)
print(total_orders)

#Calculate the drink upsell percentage per restaurant
drink_upsell_percentage_per_restaurant = (drink_orders_per_restaurant / total_orders) * 100

# Calculate the cookie upsell percentage per restaurant
cookie_upsell_percentage_per_restaurant = (cookie_orders_per_restaurant / total_orders) * 100

print(drink_upsell_percentage_per_restaurant)
print(cookie_upsell_percentage_per_restaurant)





print(df_restaurants.columns)
# Merge cookie orders with restaurant names
drink_upsell_percent_reset = drink_upsell_percentage_per_restaurant.reset_index()
drink_upsell_percent_reset.columns = ['RESTAURANT_ID', 'Drink Upsell %']
drink_merged = drink_upsell_percent_reset.merge(
    df_restaurants[['RESTAURANT_ID', 'NAME']],
    on='RESTAURANT_ID',
    how='left'
)



# Merge drink orders with restaurant names
drink_orders_per_restaurant = drink_orders_per_restaurant.reset_index()
drink_orders_per_restaurant = drink_orders_per_restaurant.merge(
    df_restaurants[['RESTAURANT_ID', 'NAME']],
    on='RESTAURANT_ID',
    how='left'
)

print(cookie_orders_per_restaurant)
print(drink_orders_per_restaurant)




# Sort the drink upsell percentages by restaurant in descending order
sorted_drink_upsells = drink_upsell_percentage_per_restaurant.sort_values(ascending=False)
print(sorted_drink_upsells.head())

# Sort the cookie upsell percentages by restaurant in descending order
sorted_cookie_upsells = cookie_upsell_percentage_per_restaurant.sort_values(ascending=False)
print(sorted_cookie_upsells.head())

# Set RESTAURANT_ID as index in df_restaurants for smooth merging
df_restaurants_indexed = df_restaurants.set_index('RESTAURANT_ID')

# Merge drink upsell % directly
drink_merged = drink_upsell_percentage_per_restaurant.to_frame(name='Drink Upsell %').merge(
    df_restaurants_indexed[['NAME']],
    left_index=True,
    right_index=True
)

# Merge cookie upsell % directly
cookie_merged = cookie_upsell_percentage_per_restaurant.to_frame(name='Cookie Upsell %').merge(
    df_restaurants_indexed[['NAME']],
    left_index=True,
    right_index=True
)

print(drink_merged)
print(cookie_merged)





187396
476507
RESTAURANT_ID
R10002    33326
R10004    26713
R10006    26336
R10008    24722
R10005    22277
R10007    22161
R10003    18222
R10001    13639
dtype: int64
RESTAURANT_ID
R10004    85835
R10002    65187
R10008    62759
R10007    56246
R10005    56178
R10006    54618
R10001    49976
R10003    45708
dtype: int64
2387224
RESTAURANT_ID
R10002    1.396015
R10004    1.118998
R10006    1.103206
R10008    1.035596
R10005    0.933176
R10007    0.928317
R10003    0.763313
R10001    0.571333
dtype: float64
RESTAURANT_ID
R10004    3.595599
R10002    2.730661
R10008    2.628953
R10007    2.356126
R10005    2.353277
R10006    2.287929
R10001    2.093478
R10003    1.914693
dtype: float64
Index(['RESTAURANT_ID', 'NAME', 'ADDRESS', 'LAT', 'LONG', 'OPENING_DATE',
       'DELIVERY_START'],
      dtype='object')
RESTAURANT_ID
R10004    85835
R10002    65187
R10008    62759
R10007    56246
R10005    56178
R10006    54618
R10001    49976
R10003    45708
dtype: int64
  RESTAURANT_ID      0       

## ✅ Conclusion 15 points

In this final section, clearly and explicitly connect your analysis to your assigned business question.

> Your conclusion **must include** the following:

- **Restate your specific business question** in one sentence.
- **Explain how your analysis and visualizations directly answer that question.**
- Reference the **specific final chart, table, or summary statistic** that provides the answer.
- Comment briefly on any **key implications** for the business.
- Ensure that the referenced visualization or table is present, labeled, and interpretable by a business audience.

🎯 **Your analysis is not complete unless this section clearly demonstrates how your final output answers the business question.**

Business Question:
Do certain restaurants have higher cookie or drink upsell rates, and can this inform targeted upselling strategies?

Conclusion:

Our analysis of upsell percentages across eight restaurant locations reveals clear differences in performance, with NYU and Midtown locations leading in both cookie and drink upsells. This indicates that upselling success is location-dependent and can guide more focused strategies.

Key Table: Restaurant Upsell Performance Overview


Restaurant	Drink Upsells %	    Cookie Upsells %
NYU	              1.12%	            3.6%
Midtown	          1.4%	            2.73%
Williamsburg	    1.04%	            2.63%
Upper West Side	  0.93%	            2.36%
Flatiron          0.93%           	2.35%
Upper East Side	  1.1%	            2.29%
Columbia	        0.57%	            2.09%
Bryant Park	      0.76%	            1.91%


Key implications for the business include the opportunity to learn from high-performing locations like NYU and Midtown, which show strong performance in both drink and cookie upsells. Restaurants with lower upsell rates, such as Columbia and Bryant Park, might benefit from targeted training or new strategies to encourage more drink and cookie purchases. Understanding these differences can lead to more effective and location-specific upselling initiatives.
